alpakka-sample-file-to-elasticsearch/src/main/java/samples/javadsl/Main.java

```diff
@@ -0,0 +1,268 @@
+import akka.stream.javadsl.*;
+import org.apache.http.HttpHost;
+import org.elasticsearch.client.RestClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import samples.common.DateTimeExtractor;
+import samples.common.RunOps;
+import samples.common.RunOpsImpl;
+
+import java.nio.file.FileSystem;
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static scala.compat.java8.FutureConverters.toJava;
+
+public class Main {
+
+    private static final Logger log = LoggerFactory.getLogger(Main.class);
+
+    private ActorSystem actorSystem;
+    private Materializer materializer;
+    private RestClient elasticsearchClient;
+
+    private final RunOps runOps;
+    private final String elasticsearchAddress;
+
+    public Main(RunOps runOps) {
+        this.runOps = runOps;
+        this.elasticsearchAddress = runOps.elasticsearchAddress();
+    }
+
+    private final static Duration streamRuntime = Duration.ofSeconds(10L);
+    private final static Integer streamParallelism = 47;
+
+    private final static String indexName = "logs";
+    private final static String typeName = "_doc";
+
+    // #directory-change-source
+
+    private final static String testDataPath = "./test-data";
+    private final static String inputLogsPath = testDataPath + "/input"; // watch directory (4)
+
+    private Source<Pair<Path, DirectoryChange>, NotUsed> directoryChangesSource() {
+        final FileSystem fs = FileSystems.getDefault();
+        final Duration pollingInterval = Duration.ofSeconds(1);
+        final int maxBufferSize = 1000;
+        return DirectoryChangesSource.create(fs.getPath(inputLogsPath), pollingInterval, maxBufferSize); // source (3)
+    }
+
+    // #directory-change-source
+
+    // #tail-logs
+
+    private Flow<Pair<Path, DirectoryChange>, Source<LogLine, NotUsed>, NotUsed> tailNewLogs() {
+        return Flow.<Pair<Path, DirectoryChange>>create()
+                // only watch for file creation events (5)
+                .filter(pair -> pair.second() == DirectoryChange.Creation)
+                .map(pair -> {
+                    Path path = pair.first();
+                    log.info("File create detected: " + path.toString());
+                    // create a new `FileTailSource` and return it as a sub-stream (6)
+                    return fileTailSource(path);
+                });
+    }
+
+    private Source<LogLine, NotUsed> fileTailSource(Path path) {
+        String filename = path.getFileName().toString();
+        String directory = path.getParent().toString();
+
+        // create `FileTailSource` for a given `path` (7)
+        return FileTailSource
+                .createLines(path, 8192, Duration.ofMillis(250))
+                .map(line -> {
+                    log.debug("Parsed > " + line);
```

Comment:
Consider the 
```suggestion
                    log.debug("Parsed > {}", line);
```
notation for logging.